ERCS/erc-7995.md

```diff
@@ -0,0 +1,587 @@
+---
+eip: 7995
+title: Encrypted Data
+description: Standard for encrypted data in Ethereum contracts with typed handles.
+author: Clément Danjou (@immortal-tofu), Ghazi Ben Armor (@GBAZama), Joseph-André Turk (@jatZama), Nicolas Pasquier (@npasquie)
+discussions-to: https://ethereum-magicians.org/t/erc-7995-encrypted-data/25013
+status: Draft
+type: Standards Track
+category: ERC
+created: 2025-07-28
+requires: 712
+---
+
+## Abstract
+
+This ERC defines a standard for encrypted data in Ethereum smart contracts. It introduces typed handles for encrypted values represented as bytes32, external ciphertext types with validation proofs, and a standardized storage layout for protocol contracts. The specification includes interfaces for access control, coprocessor-assisted operations, and key management verification, enabling confidential computation and interoperability across contracts and toolchains.
```

Comment:
This does a good job of explaining what is in your proposal, but it's missing a high-level (but technical) overview of _how_ your proposal accomplishes its goals. You could quickly sketch out the expected flow of data? I feel like that would help people browsing the repository efficiently decide if your proposal is right for them.